CPP-Example.cpp

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+   char src[40];
+   char dest[100];
+   char dest2[100];
+
+   gets(src); 
+
+   strcpy(dest, src); 
+ 
+   strncpy(dest,src, strlen(dest)); //DevSkim: ignore DS111237,DS154189 
```

Comment:
[DS111237: Banned C function detected (strncpy)](https://github.com/Microsoft/DevSkim/blob/master/guidance/DS111237.md)
strncpy is dangerous, as if the source contains 'n' or more characters, it will not null terminate the destination.

*Replacement:* Use strcpy_s or strlcpy if possible. If no safe function is viable, strcpy/strncpy should be proceeded by conditional checks to verify tha that the source string will fit in the destination with a null termnator.


Suggestions:
**[1]** +   strcpy_s(dest,  strlen(dest, src)); //DevSkim: ignore DS111237,DS154189 
**[2]** +   strlcpy(dest, src,  strlen(dest)); //DevSkim: ignore DS111237,DS154189